sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/WhereStatement.cs

```diff
@@ -31,13 +31,20 @@ public WhereLogic Where()
         /// <inheritdoc/>
         public override AdtQueryBuilder Build()
         {
-            throw new InvalidOperationException("Invalid query: Missing WHERE logic.");
+            return _parent;
         }
 
         /// <inheritdoc/>
         public override string GetQueryText()
         {
-            return QueryConstants.Where;
+            string whereLogicString = _upstreamWhereLogic.GetLogicText();
+
+            if (whereLogicString.Length > 0)
```

Comment:
an optimization that could be made here since this string can be long is to use IsNullOrEmpty. This way, we don't have to count the length of the string just to know that it's not empty.

Reply:
excellent point